### PR TITLE
allow filter functions access to asset object

### PR DESCRIPTION
**Pull request by clydin**
This allows more complex custom filtering (e.g., filtering on raw URL).

### CODE REVIEW

File: src/lib/match-options.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const matchesFilter = (filename, pattern) => {
     }
 
     if (pattern instanceof Function) {
-        return pattern(filename);
+        return pattern(filename, asset);
```
Comment (sergcen) [inline src/lib/match-options.js]: we can pass only asset

Comment (sergcen) [inline src/lib/match-options.js]: what you think?

**Comment by sergcen** [comment]
check jsDoc

File: src/lib/match-options.js (hunk 1 above)
Comment (clydin) [inline src/lib/match-options.js]: Preferred.  Wasn't sure on your policy for breaking changes.

**Comment by sergcen** [comment]
and add tests :)

**Comment by clydin** [comment]
Done.

**Pull request merged**